### DISCUSSION TITLE
Error when using a property "more"

### DISCUSSION THREAD

**Discussion by zavodptica** [richardeschloss/nuxt-highcharts#63] [Q&A]
I want to use a graph "arearange", but when using the "more" property, an error appears TypeError: Cannot read properties of undefined (reading 'addEvent').

![image](https://user-images.githubusercontent.com/43552582/236239915-465f9cf3-51c1-470b-bd40-d1d70fb3fa88.png)

reproduction link [https://stackblitz.com/edit/nuxt-3-windicss-zawmxl?file=app.vue](https://stackblitz.com/edit/nuxt-3-windicss-zawmxl?file=app.vue)

Please help me fix the code.

```
<template>
  <div>
    <highchart :options="chartOptions" more />
  </div>
</template>

<script setup>
const chartOptions = reactive({
  // chart: {
  //   type: 'arearange',
  // },
  title: {
    text: 'Chart Arearange',
  },
  series: [
    {
      type: 'arearange',
      name: 'Product',
      data: [
        [3, 5],
        [5, 12],
        [4, 6],
        [1, 8],
        [4, 8],
        [5, 12],
      ],

      color: '#7cb5ec',
      marker: {
        symbol: 'circle',
        fillColor: '#fede86',
        lineWidth: 2,
        lineColor: null,
      },
    },
  ],
  xAxis: {
    categories: ['A', 'B', 'C', 'D', 'E', 'F'],
  },
  yAxis: {
    minTickInterval: 1,
    min: 0,
    crosshair: true,
  },
  credits: {
    enabled: false,
  },
});
</script>
```

**Comment by richardeschloss** [comment]
I just updated one example that uses highcharts-more to use the composition-api: (working example)
https://github.com/richardeschloss/nuxt-highcharts/blob/feat/composition-api/components/charts/BubbleChart.vue

I'll try again with the AreaRange chart. I think the chart type needs to be specified (uncommented)

**Comment by richardeschloss** [comment]
Working example with AreaRange and "more" feature:
https://github.com/richardeschloss/nuxt-highcharts/blob/feat/composition-api/components/charts/AreaRange.vue

**Reply by zavodptica** [reply]
Thanks for your help. But when I try to repeat your code, the error remains. 
[Reproduction link on stackblitz](https://stackblitz.com/edit/nuxt-highcharts-area-range?file=components/AreaRangeChart.vue)

![image](https://user-images.githubusercontent.com/43552582/236389537-1dd89ca2-f809-400e-b075-44ee1eea8c3b.png)

**Reply by zavodptica** [reply]
The error appears in development mode. If you build the project, everything works. Apparently this is a typescript error.

![image](https://user-images.githubusercontent.com/43552582/236444817-4965d7fd-3650-4d95-95cc-72860324eaa4.png)

**Reply by richardeschloss** [reply]
Hi, thanks for bringing this to my attention.

I've only tested locally not in Stackblitz. Hmm.. but when I try right now in Stackblitz, I see another error
"vue.js?v=d3e9413e:1383 [Vue warn]: Unhandled error during execution of mounted hook " and it's bit tough to troubleshoot that in Stackblitz because I can't inspect the node_modules. But I'll see what I can do when I get the time.

**Reply by richardeschloss** [reply]
I believe the issue stems from here: https://github.com/richardeschloss/nuxt-highcharts/blob/f125a72a1d7203122912f37f4e3fd237824cd7be/lib/components.js#L207

Where for some reason in Stackblitz, HC is undefined but it shouldn't be because it's being imported at the top. 

Nonetheless, I am able to get an example working by importing highcharts and highcharts-more at the component level. It's only a few lines of boilerplate.

Working with both the Composition and Options API: 
https://stackblitz.com/edit/nuxt-highcharts-area-range-nj8xzd?file=components%2FAreaRangeChart2.vue,components%2FAreaRangeChart.vue,app.vue

**Reply by namvvo** [reply]
Cannot read properties of undefined (reading 'Core/Chart/Chart.js')  your stackblitz causes error.  It works if I just load your lib instead of using the npm module

**Reply by richardeschloss** [reply]
It appears to be due to the fact that Highcharts is primarily a client-only library (requiring a browser). Setting ssr mode to false (or making the components run client-only) seem to work: 
https://stackblitz.com/edit/nuxt-highcharts-area-range-8yvdao?file=components%2FAreaRangeChart2.vue,components%2FAreaRangeChart.vue,nuxt.config.ts
